### PR TITLE
[FIX] lunch: prevent indexerror when deleting topping

### PR DESCRIPTION
**Pull request by reka-odoo**
When the user tries to delete the toppings,
a traceback appears.

Steps to reproduce the error:
- Go to Lunch > Configuration > Vendors > Open any vendor
- Now add extra 2 (topping 2) > Save
- Delete that extra 2 (topping 2) > Save

Traceback:
```
IndexError: list index out of range
  File "odoo/http.py", line 2251, in __call__
    response = request._serve_db()
  File "odoo/http.py", line 1827, in _serve_db
    return self._transactioning(_serve_ir_http, readonly=ro)
  File "odoo/http.py", line 1847, in _transactioning
    return service_model.retrying(func, env=self.env)
  File "odoo/service/model.py", line 134, in retrying
    result = func()
  File "odoo/http.py", line 1825, in _serve_ir_http
    return self._serve_ir_http(rule, args)
  File "odoo/http.py", line 1832, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "odoo/http.py", line 2057, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "odoo/addons/base/models/ir_http.py", line 220, in _dispatch
    result = endpoint(**request.params)
  File "odoo/http.py", line 739, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "addons/web/controllers/dataset.py", line 38, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "addons/web/controllers/dataset.py", line 34, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "odoo/api.py", line 458, in call_kw
    result = getattr(recs, name)(*args, **kwargs)
  File "addons/web/models/models.py", line 71, in web_save
    self.write(vals)
  File "addons/lunch/models/lunch_supplier.py", line 193, in write
    topping_values = topping[2]
```

https://github.com/odoo/odoo/blob/07f6d71e4dd3fac7b4cacc545819430aa48823cd/addons/lunch/models/lunch_supplier.py#L192-L193 Here when the user deletes the extra 2 (topping 2),
"topping" will be [2,2] instead of [2,2,False].
So when it tries to access topping[2],
It will lead to the above traceback.

sentry-5203693767

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by reka-odoo** [comment]
Hello @odoo/sentry_reviewers :wave: 
Please review this PR.
Thanks!

File: addons/lunch/models/lunch_supplier.py [hunk 1]
```diff
@@ -190,11 +190,11 @@ def create(self, vals_list):
 
     def write(self, values):
         for topping in values.get('topping_ids_2', []):
-            topping_values = topping[2]
+            topping_values = topping[2] if topping[0] not in [Command.UNLINK, Command.DELETE] else False
```
Comment (abla001) [inline addons/lunch/models/lunch_supplier.py:193]: I'm not sure I understand this condition, is there a reason why checking the length of `topping` is not enough ?
also can you please add a test for this fix ?
Thank you in advance :pray:

Comment (reka-odoo) [reply addons/lunch/models/lunch_supplier.py]: Hello :wave: 
I added the test case.
Thanks!

**Review by abla001 — APPROVED** [review]
LGTM

**Review by abla001 — CHANGES_REQUESTED** [review]
make the comments above the lines

**Comment by reka-odoo** [comment]
> make the comments above the lines

Done.

File: addons/lunch/tests/test_supplier.py [hunk 2]
```diff
@@ -186,3 +186,39 @@ def test_cron_sync_nextcall(self):
...
+                (0, 0, {
+                    'name': 'sugar',
+                    'price': 10,
+                    'company_id': self.env.company.id
+                }),
+            ],
+        })
+
+        supplier.write({
+            # we are simulating the delete as it's done on frontend
```
Comment (abla001) [inline addons/lunch/tests/test_supplier.py:215]: also not inside a method call :sweat_smile: , something like :
```py
# simulating the delete as it's done on frontend
supplier.write({
            'topping_ids_3': [(2, supplier.topping_ids_3.id)],
        })
```

Comment (reka-odoo) [reply addons/lunch/tests/test_supplier.py]: Done :+1:

**Comment by reka-odoo** [comment]
Hello @sofiagvaladze :wave: 
Can you review this PR?
Thanks!

**Pull request closed without merging**